gbpservice/nfp/orchestrator/drivers/orchestration_driver.py

```diff
@@ -606,26 +696,36 @@ def create_network_function_device(self, device_data,
         except Exception as e:
             self._increment_stats_counter('port_details_get_failures')
             LOG.error(_LE('Failed to fetch list of interfaces to attach'
-                ' for device creation %(error)s'), {'error': e})
+                          ' for device creation %(error)s'), {'error': e})
             self._delete_interfaces(device_data, interfaces,
                                     network_handler=network_handler)
             self._decrement_stats_counter('management_interfaces',
                                           by=len(interfaces))
             return None
 
         instance_name = device_data['name']
-        try:
-            instance_id = self.compute_handler_nova.create_instance(
-                    token, self._get_admin_tenant_id(token=token),
-                    image_id, flavor,
-                    interfaces_to_attach, instance_name)
-        except Exception as e:
+        instance_id_result = {}
+        port_details_result = {}
+
+        executor.add_job('CREATE_INSTANCE',
+                         self.create_instance,
+                         self.compute_handler_nova,
+                         token, admin_tenant_id, image_id, flavor,
+                         interfaces_to_attach, instance_name,
+                         result_store=instance_id_result)
+
+        executor.add_job('GET_NEUTRON_PORT_DETAILS',
+                         self.get_neutron_port_details,
+                         network_handler, token,
+                         management_interface['port_id'],
+                         result_store=port_details_result)
+
+        completed = executor.fire()
+
+        instance_id = instance_id_result.get('result', None)
+        if not instance_id:
             self._increment_stats_counter('instance_launch_failures')
-            LOG.error(_LE('Failed to create %(device_type)s instance.'
-                          'Error: %(error)s'),
-                      {'device_type': (
-                          device_data['service_details']['device_type']),
-                       'error': e})
+            LOG.error(_LE('Failed to create %(device_type)s instance.'))
```

Comment:
It'll be a duplicate log, 1 failure log we already print with the exception of create_instance(),
same applicable for get_image_id, get_neutron_port_details too.